src/components/wallet/NewMnemonicGenerate.tsx

```diff
@@ -1,51 +1,34 @@
 import React, { useState, useCallback } from 'react'
 
-import { crypto } from '@binance-chain/javascript-sdk'
-import { KeyStore } from '@binance-chain/javascript-sdk/typings/crypto'
+import { Client as BinanceClient } from '@thorchain/asgardex-binance'
 import { delay } from '@thorchain/asgardex-util'
 import { Form, Input, Button } from 'antd'
 import { Store } from 'antd/lib/form/interface'
 
-const NewMnemonicGenerate: React.FC<{ mnemonic: string }> = ({ mnemonic }): JSX.Element => {
+type Props = {}
+const NewMnemonicGenerate: React.FC<Props> = (_: Props): JSX.Element => {
   const [loadingMsg, setLoadingMsg] = useState<string>('')
 
-  const createMnemonicWallet = (mnemonic: string, password: string) => {
-    if (!localStorage.getItem('keystore')) {
-      const privKey: string = crypto.getPrivateKeyFromMnemonic(mnemonic)
-      const keystore: KeyStore = crypto.generateKeyStore(privKey, password)
-      // TODO: dynamically set network for client, default is testnet
-      const address: string = crypto.getAddressFromPrivateKey(privKey)
-      // Temporary store during development
-      localStorage.setItem('address', address)
-      localStorage.setItem('keystore', JSON.stringify(keystore))
-    } else {
-      throw Error('keystore already exists')
-    }
+  const createMnemonicWallet = () => {
+    const phrase = BinanceClient.generatePhrase()
+    // TODO (@Veado) Extract this into helper
+    localStorage.setItem('phrase', phrase)
   }
-  const handleFormFinish = useCallback(
-    async (vals: Store) => {
-      await delay(200)
-      try {
-        setLoadingMsg('Creating wallet...')
-        createMnemonicWallet(mnemonic, vals.password)
-      } catch (err) {
-        setLoadingMsg('')
-        console.log(err)
-      }
-    },
-    [mnemonic]
-  )
+
+  const handleFormFinish = useCallback(async (_: Store) => {
+    await delay(200)
```

Comment:
What is this delay for?

Reply:
Good catch. That's just an "artifact" from previous implementation - I will remove it.